hyperspy/drawing/_widgets/rectangles.py

```diff
@@ -55,6 +60,24 @@ def _set_patch(self):
             picker=True,)]
         super(SquareWidget, self)._set_patch()
 
+    def _onjumpclick(self, event):
+        # Callback for MPL pick event
+        if event.key == "shift" and event.inaxes:
+            self.position = (event.xdata, event.ydata)
+
+    def _pos_changed(self):
+        """Call when the position of the widget has changed. It triggers the
+        relevant events, and updates the patch position.
+        """
+        if self._navigating:
+            with self.axes_manager.events.indices_changed.suppress():
+                for i in range(len(self.axes)):
+                    self.axes[i].value = self._pos[i]
+            self.axes_manager.events.indices_changed.trigger(obj=self.axes_manager)
```

Comment:
Instead to making a `SquareWidget._pos_changed` the code should be changed in `DraggableWidgetBase._pos_changed`?